sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/BoundedWindow.java

```diff
@@ -21,11 +21,10 @@
 import org.joda.time.Instant;
 
 /**
- * A {@code BoundedWindow} represents a finite grouping of elements, with an
- * upper bound (larger timestamps represent more recent data) on the timestamps
- * of elements that can be placed in the window. This finiteness means that for
- * every window, at some point in time, all data for that window will have
- * arrived and can be processed together.
+ * A {@code BoundedWindow} represents window information assigned to data elements.
+ * It has one method {@link #maxTimestamp()} to define an upper bound(included).
+ * Elements within the timestamp range are placed in this window.
```

Comment:
Since you are editing this, I will give you all of my suggestions, even expanding it a lot.

A window technically does not have a lower bound, which is worth noting.

And the only reason an element timestamp has to be below the upper bound is because the watermark is based on element timestamps and the watermark also controls discarding data and expiring the window.